src/main/java/io/fabric8/maven/docker/service/DockerAccessFactory.java

```diff
@@ -0,0 +1,227 @@
+package io.fabric8.maven.docker.service;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import io.fabric8.maven.docker.access.DockerAccess;
+import io.fabric8.maven.docker.access.DockerConnectionDetector;
+import io.fabric8.maven.docker.access.DockerMachine;
+import io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient;
+import io.fabric8.maven.docker.config.DockerMachineConfiguration;
+import io.fabric8.maven.docker.util.EnvUtil;
+import io.fabric8.maven.docker.util.Logger;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.codehaus.plexus.component.annotations.Component;
+
+/**
+ *
+ */
+@Component(role = DockerAccessFactory.class, instantiationStrategy = "singleton")
+public class DockerAccessFactory {
+
+    // Minimal API version, independent of any feature used
+    public static final String API_VERSION = "1.18";
+
+    public DockerAccess createDockerAccess(DockerAccessContext dockerAccessContext, Logger log) throws MojoExecutionException, MojoFailureException {
```

Comment:
Guess this is what you need for f-m-p to create an own Access object, right ?

Makes much sense to me. Wonder whether we should call `DockerAccessContext` better `DockerAccessConfig` (as it is what it really is, a read-only configuration object without logic). But I'm fine with, context, too. Maybe we could even stuff the Logger into it then.